pkg/types/etc.go

```diff
@@ -83,6 +83,22 @@ func IsTypeInteger(tp byte) bool {
 	return false
 }
 
+// IsTypeStoredAsInteger returns a boolean indicating whether the tp is stored as integer type.
+func IsTypeStoredAsInteger(tp byte) bool {
+	switch tp {
+	case mysql.TypeTiny, mysql.TypeShort, mysql.TypeInt24, mysql.TypeLong, mysql.TypeLonglong:
+		return true
+	case mysql.TypeYear:
+		return true
+	// Enum and Set are stored as integer type but they can not be pushed down to TiFlash
+	// case mysql.TypeEnum, mysql.TypeSet:
+	// 	return true
+	case mysql.TypeDatetime, mysql.TypeDate, mysql.TypeTimestamp, mysql.TypeDuration:
+		return true
+	}
+	return false
```

Comment:
Why not compact these cases into one branch?

Reply:
It is more clear, Integer type and date type.